### PR TITLE
Revamp ebuilds

### PR DESCRIPTION
**Pull request by parona-source**
Dropped keywords as I suspect they were added haphazardly. I have no problem readding if you actually have such hardware and have tested it on that.

### CODE REVIEW

**Comment by parona-source** [comment]
Also an alternative view on how you could handle PATH https://gitlab.com/src_prepare/src_prepare-overlay/-/blob/master/dev-util/plan9port/plan9port-2021.10.05.ebuild?ref_type=heads#L48

**Comment by parona-source** [comment]
Also minimal -> tiny because it installs tiny programs in addition rather than being more minimal by omitting something.

**Comment by parona-source** [comment]
And /opt/bsd instead of /usr/bsd, because its less random from a FHS perspective

**Comment by parona-source** [comment]
(and yes there are no tests for real https://github.com/chimera-linux/chimerautils/actions/runs/5897935032/job/15998248788#step:5:736)

**Comment by parona-source** [comment]
BSD_PREFIX could also be /usr/lib/bsd. This would be similar to how llvm handles itself.

**Comment by Connor-GH** [comment]
The difference is that llvm is a library, whilst these utils are not. I think /opt/bsd is a good idea, but i originally wanted /usr because that's where most third-party programs get installed. /opt seems like a good fit as well though, but I'd like to look at more examples.

**Comment by parona-source** [comment]
> The difference is that llvm is a library, whilst these utils are not. I think /opt/bsd is a good idea, but i originally wanted /usr because that's where most third-party programs get installed. /opt seems like a good fit as well though, but I'd like to look at more examples.
> […](#)
> On October 27, 2023 7:53:26 PM UTC, Alfred Wingate ***@***.***> wrote: BSD_PREFIX could also be /usr/lib/bsd. This would be similar to how llvm handles itself. -- Reply to this email directly or view it on GitHub: [#1 (comment)](https://github.com/Connor-GH/bsd-overlay/pull/1#issuecomment-1783431484) You are receiving this because you are subscribed to this thread. Message ID: ***@***.***>

keep in mind that llvm is slotted so it installs *everything* in there. so binaries and headers included. with llvm I mean the umberlla under llvm-project and not just libllvm.so

**Comment by Connor-GH** [comment]
/opt:
> “reserved for the installation of add-on application software packages.” 

/usr/:
> "Secondary hierarchy for read-only user data; contains the majority of (multi-)user utilities and applications. Should be shareable and read-only."

/usr/local:
> "Tertiary hierarchy for local data, specific to this host. Typically has further subdirectories (e.g., bin, lib, share"

Those seem to be the options.

**Comment by parona-source** [comment]
https://devmanual.gentoo.org/general-concepts/filesystem/index.html

https://projects.gentoo.org/qa/policy-guide/filesystem.html#pg0201

> Furthermore, only the following subdirectories of /usr are permitted:

**Comment by parona-source** [comment]
Also system package manager shouldn't touch */local its for the system administrator to use as they please.

**Comment by Connor-GH** [comment]
I saw the explicit mention of /usr/local.

Hmm, I am not liking my options here. Different sources try and say that opt is more for things like steam or discord - binaries that typically ship all of their dependencies and do things very differently. 

/opt might be the only right option here though.

**Comment by parona-source** [comment]
Ideally it would be installed normally, but that would require app-alternatives and picking non conflicting names for the binaries/manpages.

**Comment by Connor-GH** [comment]
right. app-alternatives/{coreutils,findutils,etc} is an idea for the future, but not now.

**Comment by parona-source** [comment]
I still think /usr/lib is also valid in practise. The rule about installing libraries only isn't respected. Notable systemd, pkgconfig, python, gcc and llvm do their own thing there in their own directories.

But /opt/bsd or /opt/chimerautils is fine /opt is for the oddballs.

**Comment by Connor-GH** [comment]
I prefer /opt/bsd over /opt/chimerutils because the only thing branded about them is the name. Many files aren't touched at all from FreeBSD, for example.

**Comment by Connor-GH** [comment]
As it looks right now, I like it except for the keywords. I only tested amd64 as you guessed, but I have a few VMs for riscv and aarch64 and I could test on those. Do you have powerpc hardware/VMs?

**Comment by parona-source** [comment]
My point about keywords is to just not falsely claim "it should work" when you don't know. And its best to keyword it if there is someone with that hardware and is interested themselves.

File: sys-libs/libxo/libxo-1.6.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	virtual/pkgconfig
+"
+
+src_prepare() {
+	eapply_user
+
+	# sysctl.h isn't available anymore in glibc and the functionality was
+	# removed from the linux kernel
+	# https://sourceware.org/pipermail/glibc-cvs/2020q2/069366.html
+	sed -i -e 's|#include <sys/sysctl.h>||' libxo/xo_syslog.c || die
```
Comment (parona-source) [inline sys-libs/libxo/libxo-1.6.0-r1.ebuild:32]: https://github.com/Juniper/libxo/pull/89

Hopefully this can be removed in the next release

Comment (Connor-GH) [reply sys-libs/libxo/libxo-1.6.0-r1.ebuild:32]: I thought I put in a PR for this? I guess not. I meant to but I must've been discouraged.

**Comment by Connor-GH** [comment]
I'm going to try and spin up a few VMs and test this on some gentoo live ISOs. Do you have suggestions for a test suite? Although not strictly required, I'd like it if the test isn't building world because most of the time that's just compiling, and compiling in a cross-architecture VM is slow.

**Comment by parona-source** [comment]
I don't think you should spend time testing it, unless you are doing it because you are interested in it for the sake of it.

Its wasted effort unless you find value in it in your own way.

But a decent enough threshold is that they A) build B) tests pass if there are tests and C) they don't SEGFAULT when you execute the binaries.

There is no motivation tbh to add more keywords unless there is someone in the wild who wants it from the package.

**Comment by Connor-GH** [comment]
I could see my own personal benefit in an aarch64 build guaranteed to work as I plan on doing some related things with a laptop sooner or later.

Some subset of the GNU coreutils test suite would be something I'd consider to test every flag versus the GNU counterpart to make sure they deliver expected (but perhaps not exact) results.

**Comment by Connor-GH** [comment]
This looks good as is. I'll change keywords as needed.

**Pull request merged**